### PR TITLE
Upd: Adding Symfony 4 Compatibility

### PR DESCRIPTION
**Pull request by macroprog**
#SymfonyConHackday2017

### CODE REVIEW

**Comment by macroprog** [comment]
Unit test has to be updated to manage the new version of Propel.
That's why Travis fails

**Comment by spike31** [comment]
@macroprog 
Did you try to not change propel dependencies?
Tests are green on master branch, this is important :)

File: composer.json [hunk 1]
```diff
@@ -24,13 +24,13 @@
         "doctrine/orm": "^2.2.3",
         "doctrine/doctrine-bundle": "~1.2",
         "doctrine/data-fixtures": "~1.1",
-        "doctrine/mongodb-odm-bundle": "~3.0",
+        "doctrine/mongodb-odm-bundle": "~3.4",
         "propel/propel-bundle": "~3.0@dev",
```
Comment (dkozickis) [inline composer.json]: Maybe make this "~1.5|~3.0@dev" ?

**Comment by macroprog** [comment]
@spike31 Yes, i start my tests by keeping the propel version that is in the master branch, but i get a lot of other errors as I have do be on PHP 7, and propel 1 is not compatible with it.
I will try to go more in details with it as soon as i can

**Comment by Bukashk0zzz** [comment]
Any news?

**Comment by macroprog** [comment]
Nope, still stuck with the Propel problem, can you help please ?

**Comment by Bukashk0zzz** [comment]
If have no ideas I will try on weekend.

**Comment by eugenekurasov** [comment]
Hi,
You change propel/propel-bundle from ~1.5 to ~3.0@dev.
propel/propel-bundle ~1.5 using https://github.com/propelorm/Propel
propel/propel-bundle ~3.0@dev using https://github.com/propelorm/Propel2
Classes from https://github.com/propelorm/Propel also used in other places (For [example](https://github.com/lexik/LexikTranslationBundle/blob/master/Form/Handler/TransUnitFormHandler.php#L110)).

**Comment by macroprog** [comment]
I understand that, but the bigest problem i have for now is to replace PropelQuickBuilder::buildSchema.
I can't find a solution to replace it. Do you have an idea ?

**Comment by eugenekurasov** [comment]
I will try upgrade project to Propel2 at this weekend.

**Comment by macroprog** [comment]
Hi, Could you upgrade the project to Propel2 as expected ? Thanks

**Comment by eugenekurasov** [comment]
Hi, I was at vacation, I will create PR at this week.

**Comment by eugenekurasov** [comment]
Hi, 
I am done PR for migrate to Propel 2. I waiting merge.

**Comment by spike31** [comment]
@eugenekurasov PR merged :+1:

**Comment by eugenekurasov** [comment]
@macroprog can you continue your PR? Problems with propel is should be solved

**Comment by macroprog** [comment]
I'll do the merge on next week, i can't before

**Comment by amitlimbadia** [comment]
gr8 so when we will have sf4 LexikTranslationBundle support?

**Comment by macroprog** [comment]
Propel bundle has been released for SF3.* but not yet for SF4 (no branch dedicated to this version). 
Travis still brake down because of it on the SF4 test and the Mongo Extension.
I'm still trying to find a solution. I'll follow in the coming week.

**Comment by amitlimbadia** [comment]
Any updates regarding symfony 4 compatibility?

**Comment by eugenekurasov** [comment]
@macroprog PR (https://github.com/propelorm/PropelBundle/pull/483) - has been merge. Can you continue this PR?

**Comment by macroprog** [comment]
As i said in my last post, the propel bundle has been released for Symfony 3.4, but not Symfony 4 (problem with Mongo exntesion => [https://travis-ci.org/lexik/LexikTranslationBundle/jobs/334241571](https://travis-ci.org/lexik/LexikTranslationBundle/jobs/334241571) ).
Can you manage it on the propel bundle repo to merge it here ?

**Comment by eugenekurasov** [comment]
Propel-bundle 3.0 will not support SF4. Propel-bundle already have branch 4.0 with support SF4.
I asked to create new tag in issue (https://github.com/propelorm/PropelBundle/issues/484)

**Comment by macroprog** [comment]
Okay, thank you for this information. I close this one as the other one is in green. Thank you for your help.

**Comment by spike31** [comment]
Thx @macroprog for your time and contribution

**Pull request closed without merging**